streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -459,6 +468,38 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
         }
     }
 
+    private ApplicationState getApplicationState(
```

Comment:
nit: we don't use "get" in Streams getter names. I guess this isn't exactly a pure getter, but still. On that note, perhaps a better name would be `buildApplicationState`? 🤔 

Also: even though it's all internal, I've been on a crusade to get everyone to write javadocs for methods in the StreamsPartitionAssignor with at least a brief explanation of what it does.

It's just a super complicated class that does a lot and often mutates things in a way that isn't obvious, so every little bit of documentation helps